File: typical90/d.hs
import Control.Monad
import Data.List

main = do
  [h, w] <- map read . words <$> getLine
  a <- replicateM h (map read . words <$> getLine) :: IO [[Int]]
  let memoH = [sum line | line <- a]
  let memoW = map sum $ transpose a
  let answer = [ans | (i, h) <- enumerate memoH, (j, w) <- enumerate memoW, let ans = h + w - (a !! i !! j)]
  replicateM h $ print (map (intercalate " ") answer)

enumerate = zip [0 ..]

File: typical90/d.py
H, W = list(map(int, input().split(" ")))
A = [list(map(int, input().split(" "))) for _ in range(H)]

memoH = list(map(sum, A)) # 各行の合計
memoW = [sum(column) for column in zip(*A)] # 各列の合計

answer = [[] for _ in range(H)]
for h in range(H):
    for w in range(W):
        answer[h].append(memoH[h] + memoW[w] - A[h][w])

for a in answer:
    print(" ".join(list(map(str, a))))
